src/main/java/teammates/logic/core/CoursesLogic.java

```diff
@@ -715,4 +715,13 @@ public boolean hasIndicatedSections(String courseId) throws EntityDoesNotExistEx
         return archivedCourseIds;
     }
 
+    /**
+     * Returns all courses in the Datastore.
+     * @deprecated Not scalable. Use only in admin features.
+     */
+    @Deprecated
+    public List<CourseAttributes> getAllCourses() {
```

Comment:
I feel that instead of adding production code to support client scripts, we can just use datastore calls directly in client scripts, e.g. `ofy().load().type(Course.class).list()` (and convert to attributes if necessary).

We can then work towards removing all the deprecated methods used only in client scripts from production code. Not sure what @damithc feels about this though.